update_test.go

```diff
@@ -10,7 +10,8 @@ func TestDynamicUpdateParsing(t *testing.T) {
 	for _, typ := range TypeToString {
 		if typ == "OPT" || typ == "AXFR" || typ == "IXFR" || typ == "ANY" || typ == "TKEY" ||
 			typ == "TSIG" || typ == "ISDN" || typ == "UNSPEC" || typ == "NULL" || typ == "ATMA" ||
-			typ == "Reserved" || typ == "None" || typ == "NXT" || typ == "MAILB" || typ == "MAILA" {
+			typ == "Reserved" || typ == "None" || typ == "NXT" || typ == "MAILB" || typ == "MAILA" ||
+			typ == "UINFO" {
```

Comment:
Tests were failing, because `setUINFO` uses `endingToTxtSlice` and assumes that the resulting slice is non-empty. I figured this was the most straightforward fix, since [UINFO](http://www.iana.org/assignments/dns-parameters/dns-parameters.xhtml#dns-parameters-4) is an "IANA-Reserved" nonstandard type anyway. But if you prefer, I could instead update `setUINFO`.